### ISSUE TITLE
Change `numeric-separators-style` after decimal point

### ISSUE THREAD

**Issue by oscarhermoso** [sindresorhus/eslint-plugin-unicorn#2283]
Currently, `unicorn/numeric-separators-style` reports an error for this code:

```js
// hot code path, precalculate 4 (sqrt(2) - 1) / 3
const kappa = 0.55228474983;
//            ^^^^^^^^^^^^^
//            Invalid group length in numeric value.eslint
//            [unicorn/numeric-separators-style](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v49.0.0/docs/rules/numeric-separators-style.md)
```

### Current behavior

With the default config, this is how the rule expects the number to be formatted:

```js
const kappa = 0.552_284_749_83; // not very readable
```

_In my personal opinion_, the decimal point is "lost" amongst the underscores

I could set a `groupLength` of 5...

```js
const kappa = 0.55228_47498_3; // Wikipedia style - readable
```

...but this gets in the way of formatting large numbers

```js
const one_million = 10_00000; // uh-oh, very misleading
```

### Improvement

This can be resolved by having a different `groupLength` before and after the decimal place. For example, Wikipedia formats their numbers like this:

> 149 597 870 700 metres is 1 [astronomical unit](https://en.wikipedia.org/wiki/Astronomical_unit)
3.14159 26535 89793 23846 is [π](https://en.wikipedia.org/wiki/Pi) rounded to 20 decimal places
2.71828 18284 59045 23536 is [e](https://en.wikipedia.org/wiki/E_(mathematical_constant)) rounded to 20 decimal places.

https://en.wikipedia.org/wiki/Decimal_separator#Data_versus_mask

```js
const kappa = 0.55228_47498_3; 

const one_million = 1_000_000;
```

Alternatively, it would be fine to disable checking of `numeric-separators-style` after the decimal point

```js
const kappa = 0.55228474983; // totally acceptable

const one_million = 1_000_000;
```

**Comment by sindresorhus** [comment]
I agree. I also see little value from forcing separators for the fractional part. I would go with `fractionGroupLength` and default it to `Infinity` (alternatively a separate option to not enforce this for the fraction part).

**Comment by fregante** [comment]
> precalculate 4 (sqrt(2) - 1) / 3

Is this piece of code run on Node or in the browser? I think some/most minifiers will attempt to resolve simple static calculations, simply to save bytes.

**Comment by balazser** [comment]
I like the idea. Initially, I wanted to ask if it's possible to switch it off entirely, but I found it more useful to set `fractionGroupLength` to 0. :handshake:
